nilearn/decoding/tests/test_decoder.py

```diff
@@ -1144,6 +1156,40 @@ def test_decoder_screening_percentile_surface(perc, _make_surface_class_data):
         assert model.screening_percentile_ == perc
 
 
+@pytest.mark.parametrize("screening_percentile", [30, 80, 100])
+def test_decoder_screening_percentile_adjustment_surface(
+    _make_surface_mask, _make_surface_class_data, screening_percentile
+):
+    """Test adjustment of screening percentile, such that:
+    - if mask size is smaller than screening percentile wrt to the mesh size,
+    use whole mask.
+    - if mask size is greater than screening percentile wrt to the mesh size,
+    adjust the screening percentile for that mask size.
+    """
```

Comment:
Am gonna be annoying but do you think you could split that in 2 different tests: one for each condition.

Given that the number of vertices is hard coded in the fixture, it is hard from reading the test to know when `mask_to_mesh_ratio <= screening_percentile` or not.

Reply:
Done!